src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/IVulnerable.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using NuGet.Protocol;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI
+{
+    interface IVulnerable
+    {
+        public IEnumerable<PackageVulnerabilityMetadataContextInfo> Vulnerabilities { get; }
```

Comment:
Can you use more concrete types to avoid paying the enumerator costs?

Reply:
After discussing offline, we agreed on using `IReadOnlyList` to preserve immutability, have improved Count performance, and provide the ability to use `NoAllocEnumerate` if used on a hot code path.